internal/settings/config_provider.go

```diff
@@ -83,41 +83,88 @@ func (c *configProvider) GetExcludePrefixesForDir(dirPath string) ([]string, err
 	return getExcludePrefixesForDir(dirPath)
 }
 
-// getFilePathForDir tries to find a file named DefaultConfigFilename starting in the
+// getFilePathForDir tries to find a file named by one of the ConfigFilenames starting in the
 // given directory, and going up a directory until hitting root.
 //
 // The directory must be an absolute path.
 //
 // If no such file is found, "" is returned.
-// Also returns all the directories this Config applies to.
-func getFilePathForDir(dirPath string) (string, []string) {
-	var dirPaths []string
+// If multiple files named by one of the ConfigFilenames are found in the same
+// directory, error is returned.
+func getFilePathForDir(dirPath string) (string, error) {
 	for {
-		dirPaths = append(dirPaths, dirPath)
-		filePath := filepath.Join(dirPath, DefaultConfigFilename)
-		if _, err := os.Stat(filePath); err == nil {
-			return filePath, dirPaths
+		filePath, err := getSingleFilePathForDir(dirPath)
+		if err != nil {
+			return "", err
+		}
+		if filePath != "" {
+			return filePath, nil
 		}
 		if dirPath == "/" {
-			return "", dirPaths
+			return "", nil
 		}
 		dirPath = filepath.Dir(dirPath)
 	}
 }
 
+// getSingleFilePathForDir gets the file named by one of the ConfigFilenames in the
+// given directory. Having multiple such files results in an error being returned. If no file is
+// found, this returns "".
+func getSingleFilePathForDir(dirPath string) (string, error) {
+	var filePaths []string
+	for _, configFilename := range ConfigFilenames {
+		filePath := filepath.Join(dirPath, configFilename)
+		if _, err := os.Stat(filePath); err == nil {
+			filePaths = append(filePaths, filePath)
+		}
+	}
+	switch len(filePaths) {
+	case 0:
+		return "", nil
+	case 1:
+		return filePaths[0], nil
+	default:
+		return "", fmt.Errorf("multiple configuration files in the same directory: %v", filePaths)
+	}
+}
+
 // get reads the config at the given path.
 //
-// This is expected to be in YAML format.
+// This is expected to be in YAML or JSON format, which is denoted by the file extension.
 func get(filePath string) (Config, error) {
-	data, err := ioutil.ReadFile(filePath)
+	externalConfig, err := getExternalConfig(filePath)
 	if err != nil {
 		return Config{}, err
 	}
+	return externalConfigToConfig(externalConfig, filepath.Dir(filePath))
+}
+
+func getExternalConfig(filePath string) (ExternalConfig, error) {
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return ExternalConfig{}, err
+	}
 	externalConfig := ExternalConfig{}
-	if err := yaml.UnmarshalStrict(data, &externalConfig); err != nil {
-		return Config{}, err
+	switch filepath.Ext(filePath) {
+	case ".json":
```

Comment:
Formatting nit to reduce duplication: let's lift the `len(data)` checks above the `switch` and have one `return` after the switch statement (leaving the `default` case as is).

Reply:
We can go through and refactor this ,but this would allow an empty file i.e. `prototool.toml` to produce no error with the current logic. If there's a cleanup refactor on this, let's do it in another PR, but I think the logic should stay as is.

Reply:
Should we really restrict people from having a `prototool.foo` in some directory? I _think_ this makes sense here to actually silently fail. 

It may actually be beneficial to allow a file like `prototool.sh` in a repo for running common commands. Now that I'm thinking about it, this may actually break some current users. Thoughts?

cc @amckinney

Reply:
Good point - I don’t think this actually breaks anyone (the above code), but decent point on not switching until sure.

Reply:
Moved the `len(data) == 0` check to the top, all comments should be addressed now